plugins/cloud-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeServiceImpl.java

```diff
@@ -90,6 +96,38 @@ public InstanceList run() throws Exception {
         return instances;
     }
 
+    @Override
+    public String networkAddress(String gceName) throws IOException {
+        String urlMetadataNetwork = GCE_METADATA_URL + "/" + gceName;
+        logger.debug("get network information for [{}]", gceName);
+        URL url = new URL(urlMetadataNetwork);
+        HttpHeaders headers;
+        try {
+            // hack around code messiness in GCE code
+            // TODO: get this fixed
+            headers = AccessController.doPrivileged(new PrivilegedExceptionAction<HttpHeaders>() {
```

Comment:
@rmuir are we OK with this? Is there a better way (not sure at all, just double checking).